tests/xdc/history_replication_conflict_test.go

```diff
@@ -0,0 +1,391 @@
+	2 WorkflowTaskScheduled
+	3 WorkflowExecutionSignaled {"Input": {"Payloads": [{"Data": "\"cluster2-signal\""}]}}
+	`, s.getHistory(ctx, s.cluster2, runId), []int{1, 1, 2})
+
+	// Execute pending history replication tasks. Both clusters believe they are active, therefore each cluster sends
+	// its signal to the other, triggering conflict resolution.
+	s.executeHistoryReplicationTasksUntil(ctx, "cluster1", enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED)
+	s.executeHistoryReplicationTasksUntil(ctx, "cluster2", enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED)
+
+	// cluster1 has received a signal with failover version 2 which superseded its own signal.
+	s.HistoryRequire.EqualHistoryEventsAndVersions(`
+	1 WorkflowExecutionStarted
+	2 WorkflowTaskScheduled
+	3 WorkflowExecutionSignaled {"Input": {"Payloads": [{"Data": "\"cluster2-signal\""}]}}
+	`, s.getHistory(ctx, s.cluster1, runId), []int{1, 1, 2})
+
+	// cluster2 has reapplied the signal from cluster 1 on top of its own signal.
+	s.HistoryRequire.EqualHistoryEventsAndVersions(`
+	1 WorkflowExecutionStarted
+	2 WorkflowTaskScheduled
+	3 WorkflowExecutionSignaled {"Input": {"Payloads": [{"Data": "\"cluster2-signal\""}]}}
+	4 WorkflowExecutionSignaled {"Input": {"Payloads": [{"Data": "\"cluster1-signal\""}]}}
+	`, s.getHistory(ctx, s.cluster2, runId), []int{1, 1, 2, 2})
+}
+
+func (s *historyReplicationConflictTestSuite) startWorkflowAndEnterSplitBrainState(ctx context.Context, sdkClient1 sdkclient.Client) string {
+	s.registerGlobalNamespace(ctx)
+
+	// Namespace event replication tasks are being captured; we need to execute the pending ones now to propagate the
+	// new namespace to cluster 2.
+	s.executeNamespaceReplicationTasksUntil(ctx, enumsspb.NAMESPACE_OPERATION_CREATE, 1)
+
+	runId := s.startWorkflow(ctx, sdkClient1)
+	// Process history replication tasks until a task from cluster1 => cluster2 containing the initial workflow events
+	// is encountered.
+	s.executeHistoryReplicationTasksUntil(ctx, "cluster1", enumspb.EVENT_TYPE_WORKFLOW_TASK_SCHEDULED)
+
+	s.HistoryRequire.EqualHistoryEventsAndVersions(`
+	1 WorkflowExecutionStarted
+  	2 WorkflowTaskScheduled
+  	`, s.getHistory(ctx, s.cluster1, runId), []int{1, 1})
+	s.HistoryRequire.EqualHistoryEventsAndVersions(`
+	1 WorkflowExecutionStarted
+	2 WorkflowTaskScheduled
+	`, s.getHistory(ctx, s.cluster2, runId), []int{1, 1})
+
+	// We now create a "split brain" state by setting cluster2 to active. We do not execute replication tasks afterward,
+	// so cluster1 does not learn of the change.
+	s.Equal([]string{"cluster1", "cluster1"}, s.getActiveClusters(ctx))
+	s.setActive(ctx, s.cluster2, "cluster2")
+	s.Equal([]string{"cluster1", "cluster2"}, s.getActiveClusters(ctx))
+
+	s.executeNamespaceReplicationTasksUntil(ctx, enumsspb.NAMESPACE_OPERATION_UPDATE, 2)
+	// Wait for active cluster to be changed in namespace registry entry.
+	// TODO (dan) It would be nice to find a better approach.
+	time.Sleep(waitDuration)
```

Comment:
wow 2 second is enough here? I thought namespace refresh interval is longer than that.

Reply:
Yes, I think it's enough, since the tests [set](https://github.com/temporalio/temporal/blob/main/tests/dynamicconfig.go#L38) `NamespaceCacheRefreshInterval` to 1 second.